provider/azure/environprovider.go

```diff
@@ -49,6 +50,26 @@ type ProviderConfig struct {
 	// interactively create/update service principals with
 	// password credentials.
 	InteractiveCreateServicePrincipal azureauth.InteractiveCreateServicePrincipalFunc
+
+	// TokenCreateServicePrincipal is a function used to
+	// create/update service principals with an imported access
+	// token.
+	TokenCreateServicePrincipal azureauth.TokenCreateServicePrincipalFunc
+
+	// AzListAccounts is a function used to list all the account the
+	// Azure CLI is currently logged in to.
+	AzListAccounts func() ([]azurecli.Account, error)
```

Comment:
Might it not be better to have an interface type here rather than all these functions?